diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 6bb2e64906..fbdb0b26ea 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -250,6 +250,11 @@
 #endif
 #endif
 
+; Camoufox specific files
+@RESPATH@/defaults/pref/local-settings.js
+@RESPATH@/distribution/policies.json
+@RESPATH@/camoufox.cfg
+
 ; [Default Preferences]
 ; All the pref files must be part of base to prevent migration bugs
 @RESPATH@/browser/@PREF_DIR@/firefox.js
diff --git a/lw/moz.build b/lw/moz.build
index e69de29bb2..5459e4ac36 100644
--- a/lw/moz.build
+++ b/lw/moz.build
@@ -0,0 +1,14 @@
+FINAL_TARGET_FILES += [
+  "camoufox.cfg",
+  "chrome.css",
+  "properties.json"
+]
+
+FINAL_TARGET_FILES.distribution += [
+  "policies.json",
+]
+
+# local-settings does not yet end up being pacakged.
+FINAL_TARGET_FILES.defaults.pref += [
+  "local-settings.js",
+]
diff --git a/moz.build b/moz.build
index 4d0f3d6ba9..0f07f37e92 100644
--- a/moz.build
+++ b/moz.build
@@ -235,3 +235,5 @@ SPHINX_TREES["content-security"] = "docs/content-security"
 SPHINX_TREES["jsloader"] = "docs/jsloader"
 
 include("build/templates.mozbuild")
+
+DIRS += ["lw"]
\ No newline at end of file
