diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 1d69c20fb5..c34b4c6a88 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -6721,7 +6721,7 @@ const gRemoteControl = {
       "browser.chrome.disableRemoteControlCueForTests",
       false
     );
-    if (disableRemoteControlCue && Cu.isInAutomation) {
+    if (true) {
       return;
     }
 
