diff --git a/browser/app/moz.build b/browser/app/moz.build
index a23bb7812a..58942ccb13 100644
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -182,6 +182,9 @@ for icon in ("firefox", "document", "newwindow", "newtab", "pbmode", "document_p
         icon,
     )
 
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
 if CONFIG["MOZ_ASAN"] or CONFIG["MOZ_DEBUG"]:
     WINCONSOLE = True
 else:
diff --git a/dom/base/Element.cpp b/dom/base/Element.cpp
index 101590d8c8..f004930b20 100644
--- a/dom/base/Element.cpp
+++ b/dom/base/Element.cpp
@@ -12,6 +12,8 @@
 
 #include "mozilla/dom/Element.h"
 
+#include "MaskConfig.hpp"
+
 #include <inttypes.h>
 
 #include <cstddef>
@@ -1006,6 +1008,18 @@ nsRect Element::GetClientAreaRect() {
   Document* doc = OwnerDoc();
   nsPresContext* presContext = doc->GetPresContext();
 
+  if (doc->GetBodyElement() == this) {
+    if (auto conf = MaskConfig::GetInt32Rect(
+            "document.body.clientLeft", "document.body.clientTop",
+            "document.body.clientWidth", "document.body.clientHeight")) {
+      if (conf.has_value()) {
+        auto values = conf.value();
+        return nsRect(values[0] * 60, values[1] * 60, values[2] * 60,
+                      values[3] * 60);
+      }
+    }
+  }
+
   // We can avoid a layout flush if this is the scrolling element of the
   // document, we have overlay scrollbars, and we aren't embedded in another
   // document
diff --git a/dom/base/moz.build b/dom/base/moz.build
index cd9090cda3..37a053579e 100644
--- a/dom/base/moz.build
+++ b/dom/base/moz.build
@@ -634,3 +634,6 @@ GeneratedFile(
         "/servo/components/style/properties/counted_unknown_properties.py",
     ],
 )
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/dom/base/nsGlobalWindowInner.cpp b/dom/base/nsGlobalWindowInner.cpp
index 330a707789..241f87780b 100644
--- a/dom/base/nsGlobalWindowInner.cpp
+++ b/dom/base/nsGlobalWindowInner.cpp
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "nsGlobalWindowInner.h"
+#include "MaskConfig.hpp"
 
 #include <inttypes.h>
 #include <stdio.h>
@@ -3439,6 +3440,8 @@ void nsGlobalWindowInner::SetName(const nsAString& aName,
 }
 
 double nsGlobalWindowInner::GetInnerWidth(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetDouble("window.innerWidth"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetInnerWidthOuter, (aError), aError, 0);
 }
 
@@ -3450,6 +3453,8 @@ nsresult nsGlobalWindowInner::GetInnerWidth(double* aWidth) {
 }
 
 double nsGlobalWindowInner::GetInnerHeight(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetDouble("window.innerHeight"))
+    return value.value();
   // We ignore aCallerType; we only have that argument because some other things
   // called by GetReplaceableWindowCoord need it.  If this ever changes, fix
   //   nsresult nsGlobalWindowInner::GetInnerHeight(double* aInnerWidth)
@@ -3466,12 +3471,16 @@ nsresult nsGlobalWindowInner::GetInnerHeight(double* aHeight) {
 
 int32_t nsGlobalWindowInner::GetOuterWidth(CallerType aCallerType,
                                            ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.outerWidth"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetOuterWidthOuter, (aCallerType, aError), aError,
                             0);
 }
 
 int32_t nsGlobalWindowInner::GetOuterHeight(CallerType aCallerType,
                                             ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.outerHeight"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetOuterHeightOuter, (aCallerType, aError), aError,
                             0);
 }
@@ -3486,11 +3495,13 @@ double nsGlobalWindowInner::ScreenEdgeSlopY() const {
 
 int32_t nsGlobalWindowInner::GetScreenX(CallerType aCallerType,
                                         ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.screenX")) return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScreenXOuter, (aCallerType, aError), aError, 0);
 }
 
 int32_t nsGlobalWindowInner::GetScreenY(CallerType aCallerType,
                                         ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.screenY")) return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScreenYOuter, (aCallerType, aError), aError, 0);
 }
 
@@ -3524,6 +3535,8 @@ static nsPresContext* GetPresContextForRatio(Document* aDoc) {
 double nsGlobalWindowInner::GetDevicePixelRatio(CallerType aCallerType,
                                                 ErrorResult& aError) {
   ENSURE_ACTIVE_DOCUMENT(aError, 0.0);
+  if (auto value = MaskConfig::GetDouble("window.devicePixelRatio"))
+    return value.value();
 
   RefPtr<nsPresContext> presContext = GetPresContextForRatio(mDoc);
   if (NS_WARN_IF(!presContext)) {
@@ -3594,26 +3607,38 @@ already_AddRefed<MediaQueryList> nsGlobalWindowInner::MatchMedia(
 }
 
 int32_t nsGlobalWindowInner::GetScrollMinX(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.scrollMinX"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollBoundaryOuter, (eSideLeft), aError, 0);
 }
 
 int32_t nsGlobalWindowInner::GetScrollMinY(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.scrollMinY"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollBoundaryOuter, (eSideTop), aError, 0);
 }
 
 int32_t nsGlobalWindowInner::GetScrollMaxX(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.scrollMaxX"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollBoundaryOuter, (eSideRight), aError, 0);
 }
 
 int32_t nsGlobalWindowInner::GetScrollMaxY(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetInt32("window.scrollMaxY"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollBoundaryOuter, (eSideBottom), aError, 0);
 }
 
 double nsGlobalWindowInner::GetScrollX(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetDouble("screen.pageXOffset"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollXOuter, (), aError, 0);
 }
 
 double nsGlobalWindowInner::GetScrollY(ErrorResult& aError) {
+  if (auto value = MaskConfig::GetDouble("screen.pageYOffset"))
+    return value.value();
   FORWARD_TO_OUTER_OR_THROW(GetScrollYOuter, (), aError, 0);
 }
 
diff --git a/dom/base/nsHistory.cpp b/dom/base/nsHistory.cpp
index e2fd8e6389..e5ddf4e08c 100644
--- a/dom/base/nsHistory.cpp
+++ b/dom/base/nsHistory.cpp
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "nsHistory.h"
+#include "MaskConfig.hpp"
 
 #include "jsapi.h"
 #include "mozilla/RefPtr.h"
@@ -64,6 +65,8 @@ JSObject* nsHistory::WrapObject(JSContext* aCx,
 }
 
 uint32_t nsHistory::GetLength(ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetUint32("window.history.length"))
+    return value.value();
   nsCOMPtr<nsPIDOMWindowInner> win(do_QueryReferent(mInnerWindow));
   if (!win || !win->HasActiveDocument()) {
     aRv.Throw(NS_ERROR_DOM_SECURITY_ERR);
diff --git a/dom/base/nsScreen.cpp b/dom/base/nsScreen.cpp
index 306ab35772..7f7728a6cc 100644
--- a/dom/base/nsScreen.cpp
+++ b/dom/base/nsScreen.cpp
@@ -6,6 +6,8 @@
 
 #include "nsScreen.h"
 
+#include "MaskConfig.hpp"
+
 #include "mozilla/GeckoBindings.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/DocumentInlines.h"
@@ -40,6 +42,10 @@ NS_IMPL_CYCLE_COLLECTION_INHERITED(nsScreen, DOMEventTargetHelper,
                                    mScreenOrientation)
 
 int32_t nsScreen::PixelDepth() {
+  if (auto value = MaskConfig::GetUint32("screen.colorDepth"))
+    return value.value();
+  if (auto value = MaskConfig::GetUint32("screen.pixelDepth"))
+    return value.value();
   // Return 24 to prevent fingerprinting.
   if (ShouldResistFingerprinting(RFPTarget::ScreenPixelDepth)) {
     return 24;
@@ -89,6 +95,12 @@ CSSIntRect nsScreen::GetRect() {
 }
 
 CSSIntRect nsScreen::GetAvailRect() {
+  auto rect = MaskConfig::GetRect("screen.availLeft", "screen.availTop",
+                                  "screen.availWidth", "screen.availHeight");
+  if (rect.has_value()) {
+    auto values = rect.value();
+    return {values[0], values[1], values[2], values[3]};
+  }
   // Return window inner rect to prevent fingerprinting.
   if (ShouldResistFingerprinting(RFPTarget::ScreenAvailRect)) {
     return GetTopWindowInnerRectForRFP();
diff --git a/dom/battery/BatteryManager.cpp b/dom/battery/BatteryManager.cpp
index 6322093fd9..264cc91bd1 100644
--- a/dom/battery/BatteryManager.cpp
+++ b/dom/battery/BatteryManager.cpp
@@ -9,6 +9,7 @@
 #include <cmath>
 #include <limits>
 
+#include "MaskConfig.hpp"
 #include "Constants.h"
 #include "mozilla/DOMEventTargetHelper.h"
 #include "mozilla/Hal.h"
@@ -53,6 +54,9 @@ JSObject* BatteryManager::WrapObject(JSContext* aCx,
 
 bool BatteryManager::Charging() const {
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetBool("battery:charging"); value.has_value())
+    return value.value();
+
   // For testing, unable to report the battery status information
   if (Preferences::GetBool("dom.battery.test.default", false)) {
     return true;
@@ -69,6 +73,8 @@ bool BatteryManager::Charging() const {
 
 double BatteryManager::DischargingTime() const {
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetDouble("battery:dischargingTime"))
+    return value.value();
   // For testing, unable to report the battery status information
   if (Preferences::GetBool("dom.battery.test.default", false)) {
     return std::numeric_limits<double>::infinity();
@@ -86,6 +92,8 @@ double BatteryManager::DischargingTime() const {
 
 double BatteryManager::ChargingTime() const {
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetDouble("battery:chargingTime"))
+    return value.value();
   // For testing, unable to report the battery status information
   if (Preferences::GetBool("dom.battery.test.default", false)) {
     return 0.0;
@@ -103,6 +111,7 @@ double BatteryManager::ChargingTime() const {
 
 double BatteryManager::Level() const {
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetDouble("battery:level")) return value.value();
   // For testing, unable to report the battery status information
   if (Preferences::GetBool("dom.battery.test.default")) {
     return 1.0;
diff --git a/dom/battery/moz.build b/dom/battery/moz.build
index 3a90c93c01..91d673039b 100644
--- a/dom/battery/moz.build
+++ b/dom/battery/moz.build
@@ -21,3 +21,6 @@ FINAL_LIBRARY = "xul"
 
 MOCHITEST_CHROME_MANIFESTS += ["test/chrome.toml"]
 MOCHITEST_MANIFESTS += ["test/mochitest.toml"]
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/dom/workers/WorkerNavigator.cpp b/dom/workers/WorkerNavigator.cpp
index 622724b529..c9a65ff013 100644
--- a/dom/workers/WorkerNavigator.cpp
+++ b/dom/workers/WorkerNavigator.cpp
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/dom/WorkerNavigator.h"
+#include "MaskConfig.hpp"
 
 #include <utility>
 
@@ -103,6 +104,9 @@ JSObject* WorkerNavigator::WrapObject(JSContext* aCx,
 }
 
 bool WorkerNavigator::GlobalPrivacyControl() const {
+  if (auto value = MaskConfig::GetBool("navigator.globalPrivacyControl");
+      value.has_value())
+    return value.value();
   bool gpcStatus = StaticPrefs::privacy_globalprivacycontrol_enabled();
   if (!gpcStatus) {
     JSObject* jso = GetWrapper();
@@ -125,6 +129,8 @@ void WorkerNavigator::SetLanguages(const nsTArray<nsString>& aLanguages) {
 void WorkerNavigator::GetAppVersion(nsString& aAppVersion,
                                     CallerType aCallerType,
                                     ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.appVersion"))
+    return aAppVersion.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -147,6 +153,8 @@ void WorkerNavigator::GetAppVersion(nsString& aAppVersion,
 
 void WorkerNavigator::GetPlatform(nsString& aPlatform, CallerType aCallerType,
                                   ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.platform"))
+    return aPlatform.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -207,6 +215,8 @@ class GetUserAgentRunnable final : public WorkerMainThreadRunnable {
 
 void WorkerNavigator::GetUserAgent(nsString& aUserAgent, CallerType aCallerType,
                                    ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.userAgent"))
+    return aUserAgent.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -218,6 +228,8 @@ void WorkerNavigator::GetUserAgent(nsString& aUserAgent, CallerType aCallerType,
 }
 
 uint64_t WorkerNavigator::HardwareConcurrency() const {
+  if (auto value = MaskConfig::GetUint64("navigator.hardwareConcurrency"))
+    return value.value();
   RuntimeService* rts = RuntimeService::GetService();
   MOZ_ASSERT(rts);
 
diff --git a/dom/workers/moz.build b/dom/workers/moz.build
index 2f2948a729..b8ad10403f 100644
--- a/dom/workers/moz.build
+++ b/dom/workers/moz.build
@@ -114,3 +114,6 @@ MARIONETTE_MANIFESTS += ["test/marionette/manifest.toml"]
 XPCSHELL_TESTS_MANIFESTS += ["test/xpcshell/xpcshell.toml"]
 
 BROWSER_CHROME_MANIFESTS += ["test/browser.toml"]
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
