diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 318ee7e293..8430bb4e82 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -5640,10 +5640,14 @@ nsresult XREMain::XRE_mainRun() {
 
     if (!AppShutdown::IsInOrBeyond(ShutdownPhase::AppShutdownConfirmed)) {
 #ifdef XP_MACOSX
-      if (!BackgroundTasks::IsBackgroundTaskMode()) {
+#ifdef MOZ_BACKGROUNDTASKS
+  if (!BackgroundTasks::IsBackgroundTaskMode()) {
+#endif
         rv = appStartup->CreateHiddenWindow();
         NS_ENSURE_SUCCESS(rv, NS_ERROR_FAILURE);
-      }
+#ifdef MOZ_BACKGROUNDTASKS
+  }
+#endif
 #endif
 
 #ifdef XP_WIN
