diff --git a/netwerk/protocol/http/moz.build b/netwerk/protocol/http/moz.build
index d2330003ca..a5ea856173 100644
--- a/netwerk/protocol/http/moz.build
+++ b/netwerk/protocol/http/moz.build
@@ -246,3 +246,6 @@ XPCOM_MANIFESTS += [
 ]
 
 include("/tools/fuzzing/libfuzzer-config.mozbuild")
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 83a9dcff01..2a1d18deea 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -14,6 +14,7 @@
 #include "nsError.h"
 #include "nsHttp.h"
 #include "nsHttpHandler.h"
+#include "MaskConfig.hpp"
 #include "nsHttpChannel.h"
 #include "nsHTTPCompressConv.h"
 #include "nsHttpAuthCache.h"
@@ -966,6 +967,15 @@ uint8_t nsHttpHandler::UrgencyFromCoSFlags(uint32_t cos,
 //-----------------------------------------------------------------------------
 
 const nsCString& nsHttpHandler::UserAgent(bool aShouldResistFingerprinting) {
+  if (auto value = MaskConfig::GetString("headers.User-Agent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+  if (auto value = MaskConfig::GetString("navigator.userAgent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+
   if (aShouldResistFingerprinting && !mSpoofedUserAgent.IsEmpty()) {
     LOG(("using spoofed userAgent : %s\n", mSpoofedUserAgent.get()));
     return mSpoofedUserAgent;
@@ -2082,6 +2092,10 @@ nsresult nsHttpHandler::SetAcceptLanguages() {
   }
 
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetString("headers.Accept-Language")) {
+    mAcceptLanguages.Assign(nsCString(value.value().c_str()));
+    return NS_OK;
+  }
 
   mAcceptLanguagesIsDirty = false;
 
@@ -2101,6 +2115,10 @@ nsresult nsHttpHandler::SetAcceptEncodings(const char* aAcceptEncodings,
   if (isDictionary) {
     mDictionaryAcceptEncodings = aAcceptEncodings;
   } else if (isSecure) {
+    if (auto value = MaskConfig::GetString("headers.Accept-Encoding")) {
+      mHttpsAcceptEncodings.Assign(nsCString(value.value().c_str()));
+      return NS_OK;
+    }
     mHttpsAcceptEncodings = aAcceptEncodings;
   } else {
     // use legacy list if a secure override is not specified
