diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index a60831d930..e8d5d9581e 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -379,18 +379,6 @@
         supportPage: "media-keyboard-control",
       },
       {
-        id: "cfrRecommendations",
-        l10nId: "browsing-cfr-recommendations",
-        supportPage: "extensionrecommendations",
-        subcategory: "cfraddons",
-      },
-      {
-        id: "cfrRecommendations-features",
-        l10nId: "browsing-cfr-features",
-        supportPage: "extensionrecommendations",
-        subcategory: "cfrfeatures",
-      },
-      {
         id: "linkPreviewEnabled",
         l10nId: "link-preview-settings-enable",
         subcategory: "link-preview",
