diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index 35e044bd84..bfb6fbb8b2 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -646,7 +646,6 @@ function init_all() {
     gotoPref(event.target.view);
   });
 
-  maybeDisplayPoliciesNotice();
   maybeDisplayTLSKeyLoggingNotice();
 
   window.addEventListener("hashchange", onHashChange);
