--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -543,7 +543,7 @@ urlbar-result-action-search-tabs = Searc
 # A label shown above the "Firefox Suggest" (bookmarks/history) group in the
 # urlbar results.
 urlbar-group-firefox-suggest =
-  .label = { -firefox-suggest-brand-name }
+  .label = Suggestions
 
 # A label shown above the search suggestions group in the urlbar results. It
 # should use sentence case.
