diff --git a/dom/media/CubebUtils.cpp b/dom/media/CubebUtils.cpp
index 8b6621b4b834..a14e742c0022 100644
--- a/dom/media/CubebUtils.cpp
+++ b/dom/media/CubebUtils.cpp
@@ -42,6 +42,7 @@
 #include <thread>
 #include "CallbackThreadRegistry.h"
 #include "mozilla/StaticPrefs_media.h"
+#include "MaskConfig.hpp"
 
 #define AUDIOIPC_STACK_SIZE_DEFAULT (64 * 4096)
 
@@ -408,6 +409,8 @@ uint32_t PreferredSampleRate(bool aShouldResistFingerprinting) {
   if (sCubebForcedSampleRate) {
     return sCubebForcedSampleRate;
   }
+  if (auto value = MaskConfig::GetUint32("AudioContext:sampleRate"))
+    return value.value();
   if (aShouldResistFingerprinting) {
     return 44100;
   }
diff --git a/dom/media/moz.build b/dom/media/moz.build
index 73d244e0d2ba..92d8795137e6 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -420,3 +420,6 @@ CXXFLAGS += ["-Werror=switch"]
 FINAL_LIBRARY = "xul"
 
 SPHINX_TREES["/media"] = "docs"
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
diff --git a/dom/media/webaudio/AudioContext.cpp b/dom/media/webaudio/AudioContext.cpp
index f86a9f331edb..5d862af1f4f1 100644
--- a/dom/media/webaudio/AudioContext.cpp
+++ b/dom/media/webaudio/AudioContext.cpp
@@ -45,6 +45,7 @@
 #include "mozilla/dom/StereoPannerNodeBinding.h"
 #include "mozilla/dom/WaveShaperNodeBinding.h"
 
+#include "MaskConfig.hpp"
 #include "AudioBuffer.h"
 #include "AudioBufferSourceNode.h"
 #include "AudioChannelService.h"
@@ -555,6 +556,8 @@ double AudioContext::OutputLatency() {
   }
   // When reduceFingerprinting is enabled, return a latency figure that is
   // fixed, but plausible for the platform.
+  if (auto value = MaskConfig::GetDouble("AudioContext:outputLatency"))
+    return value.value();
   double latency_s = 0.0;
   if (mShouldResistFingerprinting) {
 #ifdef XP_MACOSX
@@ -708,6 +711,8 @@ void AudioContext::UnregisterActiveNode(AudioNode* aNode) {
 }
 
 uint32_t AudioContext::MaxChannelCount() const {
+  if (auto value = MaskConfig::GetUint32("AudioContext:maxChannelCount"))
+    return value.value();
   if (mShouldResistFingerprinting) {
     return 2;
   }
diff --git a/dom/media/webaudio/moz.build b/dom/media/webaudio/moz.build
index a2fe9486004b..6a03f45bf7e4 100644
--- a/dom/media/webaudio/moz.build
+++ b/dom/media/webaudio/moz.build
@@ -154,3 +154,6 @@ include("/ipc/chromium/chromium-config.mozbuild")
 
 FINAL_LIBRARY = "xul"
 LOCAL_INCLUDES += [".."]
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
