diff --git a/netwerk/protocol/http/moz.build b/netwerk/protocol/http/moz.build
index 1b8068f235..101ab978e2 100644
--- a/netwerk/protocol/http/moz.build
+++ b/netwerk/protocol/http/moz.build
@@ -257,3 +257,6 @@ XPCOM_MANIFESTS += [
 ]
 
 include("/tools/fuzzing/libfuzzer-config.mozbuild")
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 0192e5f6f2..a43ecf238e 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -12,6 +12,7 @@
 #include "EventTokenBucket.h"
 #include "HttpLog.h"
 #include "LoadContextInfo.h"
+#include "MaskConfig.hpp"
 #include "SerializedLoadContext.h"
 #include "TRRServiceChannel.h"
 #include "Tickler.h"
@@ -993,6 +994,15 @@ uint8_t nsHttpHandler::UrgencyFromCoSFlags(uint32_t cos,
 //-----------------------------------------------------------------------------
 
 const nsCString& nsHttpHandler::UserAgent(bool aShouldResistFingerprinting) {
+  if (auto value = MaskConfig::GetString("headers.User-Agent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+  if (auto value = MaskConfig::GetString("navigator.userAgent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+
   if (aShouldResistFingerprinting && !mSpoofedUserAgent.IsEmpty()) {
     LOG(("using spoofed userAgent : %s\n", mSpoofedUserAgent.get()));
     return mSpoofedUserAgent;
@@ -2107,6 +2117,10 @@ nsresult nsHttpHandler::SetAcceptLanguages() {
   }
 
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetString("headers.Accept-Language")) {
+    mAcceptLanguages.Assign(nsCString(value.value().c_str()));
+    return NS_OK;
+  }
 
   mAcceptLanguagesIsDirty = false;
 
@@ -2123,6 +2137,11 @@ nsresult nsHttpHandler::SetAcceptLanguages() {
 
 nsresult nsHttpHandler::SetAcceptEncodings(const char* aAcceptEncodings,
                                            bool isSecure, bool isDictionary) {
+  nsCString encodingOverride;
+  if (auto value = MaskConfig::GetString("headers.Accept-Encoding")) {
+    encodingOverride.Assign(nsCString(value.value().c_str()));
+    aAcceptEncodings = encodingOverride.get();
+  }
   if (isDictionary) {
     mDictionaryAcceptEncodings = aAcceptEncodings;
   } else if (isSecure) {
