diff --git a/browser/base/content/browser-context.inc b/browser/base/content/browser-context.inc
index 3a58888b28b4..471c238aa284 100644
--- a/browser/base/content/browser-context.inc
+++ b/browser/base/content/browser-context.inc
@@ -107,9 +107,6 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 />
-      <menuitem id="context-savelinktopocket"
-                data-l10n-id="main-context-menu-save-link-to-pocket"
-                />
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 />
@@ -264,9 +261,6 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 />
-      <menuitem id="context-pocket"
-                data-l10n-id="main-context-menu-save-to-pocket"
-                />
       <menu id="context-sendpagetodevice"
             class="sync-ui-item"
             data-l10n-id="main-context-menu-send-to-device"
